lib/guardian/topic_guardian.rb

```diff
@@ -36,10 +36,16 @@ def can_publish_topic?(topic, category)
 
   # Creating Methods
   def can_create_topic?(parent)
-    is_staff? ||
-    (user &&
-      user.trust_level >= SiteSetting.min_trust_to_create_topic.to_i &&
-      can_create_post?(parent))
+    return true if is_staff?
+
+    is_allowed = user && user.trust_level >= SiteSetting.min_trust_to_create_topic.to_i && can_create_post?(parent)
+    return false unless is_allowed
+
+    # if uncategorized topics are allowed then there is at least one category available
+    return true if SiteSetting.allow_uncategorized_topics
+
+    allowed_categories = Category.topic_create_allowed(self).pluck(:id)
```

Comment:
no need to pluck all the category ids here.

`Category.topic_create_allowed(self).limit(2).count == 2`

Reply:
Done, thanks for the tip! 👍